compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -257,16 +261,17 @@ pub fn eval_static_initializer_provider<'tcx>(
 
     let instance = ty::Instance::mono(tcx, def_id.to_def_id());
     let cid = rustc_middle::mir::interpret::GlobalId { instance, promoted: None };
-    let ecx = InterpCx::new(
+    let mut ecx = InterpCx::new(
         tcx,
         tcx.def_span(def_id),
         ty::ParamEnv::reveal_all(),
         // Statics (and promoteds inside statics) may access other statics, because unlike consts
         // they do not have to behave "as if" they were evaluated at runtime.
         CompileTimeInterpreter::new(CanAccessStatics::Yes, CheckAlignment::Error),
     );
-    let alloc_id = eval_in_interpreter(ecx, cid, true)?.alloc_id;
-    let alloc = tcx.global_alloc(alloc_id).unwrap_memory();
+    let alloc_id = eval_in_interpreter(&mut ecx, cid, true)?.alloc_id;
+    let alloc = take_static_root_alloc(&mut ecx, alloc_id);
```

Comment:
Inside `take_static_root_alloc` might be a good place to do the memory kind patching?

Reply:
That's too late. Interning and validation  are already done.

My plan was to make validation take it out and return it, and then reinsert it just for validation.

Reply:
Validation happens after interning, so how can it be already done?

Reply:
this is after validation. This is the query provider. interning and validation happens inside `eval_in_interpreter`